splitio/sync/segment.py

```diff
@@ -174,13 +175,153 @@ def synchronize_segments(self, segment_names = None, dont_wait = False):
+
+        return_flag = True
+        for segment_name in segment_names:
+            if not self.synchronize_segment(segment_name):
+                return_flag = False
+
+        return return_flag
+
+    def synchronize_segment(self, segment_name, till=None):
+        """
+        Update a segment from queue
+
+        :param segment_name: Name of the segment to update.
+        :type segment_name: str
+
+        :param till: ChangeNumber received.
+        :type till: int
+
+        :return: True if no error occurs. False otherwise.
+        :rtype: bool
+        """
+        try:
+            fetched = self._read_segment_from_json_file(segment_name)
+            if fetched == {}:
+                return False
+            fetched_sha = util._get_sha(json.dumps(fetched))
+            if not self.segment_exist_in_storage(segment_name):
+                    self._segment_sha[segment_name] = fetched_sha
+                    self._segment_storage.put(segments.from_raw(fetched))
+                    _LOGGER.debug("segment %s is added to storage", segment_name)
+            else:
+                if fetched_sha != self._segment_sha[segment_name]:
+                    self._segment_sha[segment_name] = fetched_sha
+                    if self._segment_storage.get_change_number(segment_name) <= fetched['till'] or fetched['till'] == self._DEFAULT_SEGMENT_TILL:
+                        self._segment_storage.update(
+                            segment_name,
+                            fetched['added'],
+                            fetched['removed'],
+                            fetched['till']
+                        )
+                        _LOGGER.debug("segment %s is updated", segment_name)
+        except Exception as e:
+            _LOGGER.error("Could not fetch segment: %s \n" + str(e), segment_name)
+            return False
+
+        return True
+
+    def _read_segment_from_json_file(self, filename):
+        """
+        Parse a segment and store in segment storage.
+
+        :param filename: Path of the file containing split
+        :type filename: str.
+
+        :return: Sanitized segment structure
+        :rtype: Dict
+        """
+        try:
+            with open(self._segment_folder + '/' + filename + '.json', 'r') as flo:
+                parsed = json.load(flo)
+            santitized_segment = self._sanitize_segment(parsed)
+            return santitized_segment
+        except Exception as exc:
+            raise ValueError("Error parsing file %s. Make sure it's readable." % filename) from exc
+
+    def _sanitize_segment(self, parsed):
+        """
+        Sanitize json elements.
+
+        :param parsed: segment dict
+        :type parsed: Dict
+
+        :return: sanitized segment structure dict
+        :rtype: Dict
+        """
+        if 'name' not in parsed or parsed['name'] is None:
+            _LOGGER.warning("Segment does not have [name] element, skipping")
+            return {}
```

Comment:
wouldn't be better returning None instead of empty object?

Reply:
Possible, I am checking if the returned is {} then synchronize_segments() method will return False

Reply:
fixed